src/components/parser/multiStopParser.ts

```diff
@@ -47,3 +51,32 @@ export const multiStopParser = (json: EtaPredictionXml) => {
   console.log(result);
   return result;
 };
+
+export function multiStopUnifier(
+  json: EtaPredictionXml,
+  stopBookmarks: stopBookmarkRedux
+) {
+  const result = multiStopParser(json);
+
+  // phase 2: combine a/c to stop number
+  const unifiedList: stopBookmarkWithEta[] = [];
+  for (const id of stopBookmarks.ids) {
+    unifiedList.push({
+      stopId: stopBookmarks.entities[id].stopId,
+      name: stopBookmarks.entities[id].name,
+      lines: stopBookmarks.entities[id].lines,
+      ttcId: stopBookmarks.entities[id].ttcId,
+      etas: [],
+    });
+  }
+
+  for (const item of result) {
+    const matchingStop = unifiedList.findIndex(
+      (searching) => item.stopTag === searching.ttcId
+    );
+    unifiedList[matchingStop].etas = unifiedList[matchingStop].etas
+      .concat(item.etas)
+      .sort((a, b) => a.epochTime - b.epochTime);
+  }
+  return unifiedList;
```

Comment:
Now we run two forLoop for
- fetching `stopId`, `name` ... `ttcId`
- fetching `etas`

can we do that in one forLoop?

Reply:
the current time complexity is O(n^2) (`findIndex` inside a loop), can we do the following to reduce it to O(n)?

in the first loop
- store a key-pair value {ttcId: indexOf stopBookmarks} into a dictObject

in the second loop 
- forEach item in the result, check if there is a match in the dictObject, and do your stuff if found.

spaceComplexity is still O(n)

Reply:
Continue in #71